gem/personalise/models.py

```diff
@@ -0,0 +1,130 @@
+
+
+# Force current index page to display our personalised survey
+SurveysIndexPage.subpage_types.append('personalise.PersonalisableSurvey')
+
+
+def get_personalisable_survey_content_panels():
+    """
+    Replace panel for "survey_form_fields" with
+    panel pointing to the custom form field model.
+    """
+    panels = [
+        FieldPanel('segment')
+    ]
+
+    for panel in MoloSurveyPage.content_panels:
+        if isinstance(panel, InlinePanel) and \
+                panel.relation_name == 'survey_form_fields':
+            panel = InlinePanel('personalisable_survey_form_fields',
+                                label=_('personalisable form fields'))
+        panels.append(panel)
+
+    return panels
+
+
+class PersonalisableSurvey(MoloSurveyPage):
+    """
+    Survey page that enables form fields to be segmented with
+    wagtail-personalisation.
+    """
+    segment = models.ForeignKey('wagtail_personalisation.Segment',
+                                on_delete=models.SET_NULL, blank=True,
+                                null=True,
+                                help_text=_('Leave it empty to show this survey'
+                                            'to every user.'))
+    content_panels = get_personalisable_survey_content_panels()
+    template = MoloSurveyPage.template
+
+    class Meta:
+        verbose_name = _('personalisable survey')
+
+    def get_form_fields(self):
+        """Get form fields for particular segments."""
+        # Get only segmented form fields if serve() has been called
+        # (because the page is being seen by user on the front-end)
+        if hasattr(self, 'request'):
+            user_segments_ids = [s.id for s in get_segment_adapter(
+                self.request).get_segments()]
+
+            return self.personalisable_survey_form_fields.filter(
+                Q(segment=None) | Q(segment_id__in=user_segments_ids)
+            )
+
+        # Return all form fields if there's no request passed
+        # (used on the admin site so serve() will not be called).
+        return self.personalisable_survey_form_fields \
+                   .select_related('segment')
+
+    def get_data_fields(self):
+        """
+        Get survey's form field's labels with segment names
+        if there's one associated.
+        """
+        data_fields = [
+            ('created_at', _('Submission Date')),
+        ]
+
+        # Add segment name to a field label if it is segmented.
+        for field in self.get_form_fields():
+            label = field.label
+
+            if field.segment:
+                label = '%s (%s)' %(label, field.segment.name)
+
+            data_fields.append((field.clean_name, label))
+
+        return data_fields
+
+    def serve(self, request, *args, **kwargs):
+        # We need request data in self.get_form_fields() to perform
+        # segmentation.
+        #TODO(tmkn): This is quite hacky, need to come up with better solution.
```

Comment:
Hi @miltontony, the solution would need to be provided upstream in [Wagtailsurveys](https://github.com/torchbox/wagtailsurveys). It would need a `HttpRequest` object to be passed into `get_form_fields` or upstream in Wagtail where we can access request object in the `Page` class. Currently we assign it to a class property whenever serve() is called, which does not seem like a good practice and feels quite hacky, but it works flawlessly.